server/datastore/mysql/aggregated_stats.go

```diff
@@ -95,20 +95,20 @@ func (ds *Datastore) UpdateQueryAggregatedStats(ctx context.Context) error {
 
 // CalculateAggregatedPerfStatsPercentiles calculates the aggregated user/system time performance statistics for the given query.
 func (ds *Datastore) CalculateAggregatedPerfStatsPercentiles(ctx context.Context, aggregate fleet.AggregatedStatsType, queryID uint) error {
-	tx := ds.writer(ctx)
+	reader := ds.reader(ctx)
```

Comment:
Let's add a comment somewhere that this is on purpose (because we already synced reader/writer before calling `ds.CalculateAggregatedPerfStatsPercentiles `)